alea/ces_transformation.py

```diff
@@ -0,0 +1,157 @@
+from pydantic import BaseModel, validator
+from typing import Dict, Optional, Any, Literal, Callable
+import numpy as np
+from scipy import stats
+from copy import deepcopy
+from multihist import Hist1d
+
+
+def energy_res(energy, a=25.8, b=1.429):
+    """Return energy resolution in keV.
+
+    :param energy: true energy in keV :return: energy resolution in keV
+
+    """
+    # Reference for the values of a,b:
+    # xenon:xenonnt:analysis:ntsciencerun0:g1g2_update#standard_gaussian_vs_skew-gaussian_yue
+    return (np.sqrt(energy) * a + energy * b) / 100
+
+
+def smearing_mono_gaussian(
+    hist: Any,
+    smearing_a: float,
+    smearing_b: float,
+    peak_energy: float,
+    bins: Optional[np.ndarray] = None,
+):
+    """Smear a mono-energetic peak with a Gaussian."""
+
+    if bins is None:
+        # create an emptyzero histogram with the same binning as the input histogram
+        data = stats.norm.pdf(
+            hist.bin_centers,
+            loc=peak_energy,
+            scale=energy_res(peak_energy, smearing_a, smearing_b),
+        )
+        hist_smeared = Hist1d(data=np.zeros_like(data), bins=hist.bin_edges)
+        hist_smeared.histogram = data
+    else:
+        # use the bins that set by the user
+        bins = np.array(bins)
+        if bins.size <= 1:
+            raise ValueError("bins must have at least 2 elements")
+        bin_centers = 0.5 * (bins[1:] + bins[:-1])
+        data = stats.norm.pdf(
+            bin_centers,
+            loc=peak_energy,
+            scale=energy_res(peak_energy, smearing_a, smearing_b),
+        )
+        # create an empty histogram with the user-defined binning
+        hist_smeared = Hist1d(data=np.zeros_like(data), bins=bins)
+        hist_smeared.histogram = data
+
+    return hist_smeared
+
+
+def smearing_hist_gaussian(
+    hist: Any,
+    smearing_a: float,
+    smearing_b: float,
+    bins: Optional[np.ndarray] = None,
+):
+    """Smear a histogram with Gaussian. This allows for non-uniform histogram binning.
+
+    :param hist: the spectrum we want to smear :param bins: bin edges of the returned spectrum
+    :return: smeared histogram in the same unit as input spectrum
+
+    """
```

Comment:
Just for consistency in the code, please use the google-style docstring format everywhere.

Reply:
Sure will do